frame/election-provider-support/src/onchain.rs

```diff
@@ -159,13 +156,15 @@ impl<T: Config> ElectionProviderBase for OnChainExecution<T> {
 
 impl<T: Config> InstantElectionProvider for OnChainExecution<T> {
 	fn instant_elect(
-		forced_input_voters_bound: Option<u32>,
-		forced_input_target_bound: Option<u32>,
+		forced_input_voters_bounds: DataProviderBounds,
+		forced_input_targets_bounds: DataProviderBounds,
 	) -> Result<BoundedSupportsOf<Self>, Self::Error> {
-		elect_with_input_bounds::<T>(
-			Some(T::VotersBound::get().min(forced_input_voters_bound.unwrap_or(u32::MAX)) as usize),
-			Some(T::TargetsBound::get().min(forced_input_target_bound.unwrap_or(u32::MAX)) as usize),
-		)
+		let elections_bounds = ElectionBoundsBuilder::from(T::Bounds::get())
+			.voters_or_lower(forced_input_voters_bounds)
```

Comment:
I think similarly, if you implement `cmp` for `DataProviderBounds`, you can do: 

```
let elections_bounds = ElectionBoundsBuilder::from(T::Bounds::get().min(forced_bounds))
```

which also implies the input of the function could be simplified into one `ElectionBounds` rather than two `DataProviderBounds`.

Reply:
The API is overall still very confusing :( let's make a ticket to improve?